docs/docs/user-guide/stored-role-passwords.md

```diff
@@ -0,0 +1,21 @@
+# Stored Role Passwords
+
+Mathesar stores passwords for any [roles](./roles.md) that you would like to use when authenticating with PostgreSQL to work with data. Roles with stored passwords can then be assigned to [users](./users.md) via [collaborators](./collaborators.md)
```

Comment:
The phrase "authenticating with PostgreSQL to work with data." of course makes sense, but may be difficult to connect to the act of using Mathesar for most users. When does "authenticating with PostgreSQL to work with data." happen? When I connect a database? The phrase describes an "under the hood" mechanism of how Mathesar works.

I do suspect the ultimate fix here is that "Stored Role Passwords" are ultimately a field in the "add database connection" and "add collaborator" forms and don't even _need_ to be a "top level concept" for users (I described this a bit in a general document about permissions, I'll add that background below), but there's probably an incremental improvement to be made here while the concept still exists.

<details>
<summary>Expanded thoughts on "stored role passwords"</summary>

> Could storing role passwords in Mathesar simply be added to the "add collaborator" flow?
> 
> In what scenario would someone want to store a role password in Mathesar without an associated Mathesar user connected to the role?
> 
> In this new approach, the role password would be stored as it is now, but in the UI there would be a password input in the modal after clicking "add collaborator". If the role password was already stored from a previous collaborator, we could disable the input and show a "this password is already stored" message and/or let the input be editable so they could change the password if desired.
> 
> The table listing collaborators would then have an "edit collaborator" button for each collaborator, which allows for changing the associated role and/or changing the stored password.
> 
> I think this could smplify things quite a bit. The concept of a "Collaborator" as a link between a Mathesar user to a DB postgres role that gets authenticated via the role password seems to be a simpler mental model.
> 
> A small side benefit: we could potentially even remove the "In Mathesar" vs. "On The Server" concept from the settings sidebar.

</details>

Reply:
Hmmm, I'm not really sure what to make of this. It seems like you're suggesting changes to the app. For example:

> Could storing role passwords in Mathesar simply be added to the "add collaborator" flow?

This is a great suggestion. It's just that it's pretty far out of scope for this PR. In this PR the goals are: (1) docs content in the guide, and (2) minor UX and copy adjustments in the app to further align the UI and the docs with one another.

I see significant opportunity to improve the UX and iron out some of the kinks you bumped into on our call. But I suspect we probably won't take up that work right away. (That's really a question for Kriti.) And when we do, it'll involve a number of higher-level conversations between various people who were more deeply involved in all this design work, especially Pavish and Ghislaine. If I'm understanding your suggestions correctly here, I think we'll really need to wait on considering it until we open up the "access control" design can of worms again.

> In what scenario would someone want to store a role password in Mathesar without an associated Mathesar user connected to the role?

Yeah, good question. I don't think it's necessarily that someone would intentionally want this state as an end goal. But it's more so that this state is _possible_ given our data model, and may happen to occur temporarily as a user is going about other work. For example: Say I have a user `alice` assigned to the role `reporting`, and no other users assigned to that role. Alice has left the team and now I'm onboarding her replacement Bob. First I delete the user `alice`. That in turn deletes the collaborator associated with her user. Then I create Bob's user, `bob` and assign him to the `reporting` role. Mathesar keeps the stored password for `reporting` throughout this process even though there is a point where no collaborators are assigned to it.

For now, if there are docs changes you'd like to make without changing the structure of the UI, then I'd support that. But larger structural changes to the app will need to wait until after beta.